src/liballoc/lib.rs

```diff
@@ -118,6 +118,7 @@
 #![feature(unboxed_closures)]
 #![feature(unicode_internals)]
 #![feature(unsize)]
+#![cfg_attr(not(bootstrap), feature(unsized_fn_params))]
 #![feature(unsized_locals)]
```

Comment:
`unsized_locals` should not be needed any more, so can you gate this by `cfg_attr(bootstrap, ...)`?

Reply:
Unsure if I understand this comment. What do you mean? how is `unsized_locals` not needed and what does it means? should I remove feature `unsized_locals` from here?. Also I don't understand why you're suggesting to use `cfg_attr(bootstrap, ...)` ...

Reply:
We should not be using the incomplete `unsized_locals` feature in libstd bcause it currently generates incorrect code. Thus, we should remove the feature gate. But of course removing it entirely will fail so you need to instead do
```
#![cfg_attr(bootstrap, feature(unsized_locals)]
```

Reply:
With this diff ...

```
commit 742c1e57e30eaddf01a8c7029c80769cdd7e2f5d (HEAD -> separate-unsized-locals)
Author: Santiago Pastorino <spastorino@gmail.com>
Date:   Sat May 23 20:20:53 2020 -0300

    Do not "enable" feature unsized_locals in liballoc/libcore

diff --git a/src/liballoc/lib.rs b/src/liballoc/lib.rs
index 4e4d40e69d4..b00f2b4b14f 100644
--- a/src/liballoc/lib.rs
+++ b/src/liballoc/lib.rs
@@ -119,7 +119,7 @@
 #![feature(unicode_internals)]
 #![feature(unsize)]
 #![cfg_attr(not(bootstrap), feature(unsized_fn_params))]
-#![feature(unsized_locals)]
+#![cfg_attr(bootstrap, feature(unsized_locals))]
 #![feature(allocator_internals)]
 #![feature(slice_partition_dedup)]
 #![feature(maybe_uninit_extra, maybe_uninit_slice)]
diff --git a/src/libcore/lib.rs b/src/libcore/lib.rs
index 01f597fd944..cdac6a6d5bd 100644
--- a/src/libcore/lib.rs
+++ b/src/libcore/lib.rs
@@ -118,7 +118,7 @@
 #![feature(transparent_unions)]
 #![feature(unboxed_closures)]
 #![cfg_attr(not(bootstrap), feature(unsized_fn_params))]
-#![feature(unsized_locals)]
+#![cfg_attr(bootstrap, feature(unsized_locals))]
 #![feature(untagged_unions)]
 #![feature(unwind_attributes)]
 #![feature(doc_alias)]
```

I get ...

```
Building stage1 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling cc v1.0.52
   Compiling core v0.0.0 (/home/santiago/src/oss/rust2/src/libcore)
   Compiling libc v0.2.69
   Compiling autocfg v0.1.7
   Compiling std v0.0.0 (/home/santiago/src/oss/rust2/src/libstd)
   Compiling hashbrown v0.6.2
   Compiling compiler_builtins v0.1.28
   Compiling backtrace-sys v0.1.37
   Compiling unwind v0.0.0 (/home/santiago/src/oss/rust2/src/libunwind)
error[E0161]: cannot move a value of type T: the size of T cannot be statically determined
   --> src/libcore/mem/mod.rs:162:33
    |
162 |     unsafe { intrinsics::forget(t) }
    |                                 ^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0161`.
error: could not compile `core`.

To learn more, run the command again with --verbose.
command did not execute successfully: "/home/santiago/src/oss/rust2/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-unknown-linux-gnu" "-Zbinary-dep-depinfo" "-j" "8" "--release" "--features" "panic-unwind backtrace compiler-builtins-c" "--manifest-path" "/home/santiago/src/oss/rust2/src/libtest/Cargo.toml" "--message-format" "json-render-diagnostics"
expected success, got: exit code: 101
failed to run: /home/santiago/src/oss/rust2/build/bootstrap/debug/bootstrap build --stage 1 src/libstd
Build completed unsuccessfully in 0:02:00
```